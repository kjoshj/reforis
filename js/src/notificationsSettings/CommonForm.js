/*
 * Copyright (C) 2019-2025 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import { Select, CheckBox } from "foris";
import PropTypes from "prop-types";

import HELP_TEXTS from "./helpTexts";

export const SEVERITY_OPTIONS = {
    // '0': _?,
    1: _("Reboot is required"),
    2: _("Reboot or attention is required"),
    3: _("Reboot or attention is required or update was installed"),
};

CommonForm.propTypes = {
    formData: PropTypes.shape({
        to: PropTypes.string,
        severity_filter: PropTypes.oneOf(
            Object.keys(SEVERITY_OPTIONS).map((key) => parseInt(key))
        ).isRequired,
        send_news: PropTypes.bool.isRequired,
    }).isRequired,
    formErrors: PropTypes.shape({ to: PropTypes.string }),
    setFormValue: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
};

CommonForm.defaultProps = {
    setFormValue: () => {},
    formData: {},
    formErrors: {},
};

export default function CommonForm({ formData, setFormValue, disabled }) {
    return (
        <>
            <h2>{_("General Notification Settings")}</h2>
            <Select
                label={_("Importance")}
                value={formData.severity_filter}
                choices={SEVERITY_OPTIONS}
                onChange={setFormValue((value) => ({
                    emails: {
                        common: { severity_filter: { $set: parseInt(value) } },
                    },
                }))}
                helpText={HELP_TEXTS.common.severity_filter}
                disabled={disabled}
            />
            <CheckBox
                label={_("Send news")}
                checked={formData.send_news}
                helpText={HELP_TEXTS.common.send_news}
                onChange={setFormValue((value) => ({
                    emails: {
                        common: { send_news: { $set: value } },
                    },
                }))}
                disabled={disabled}
            />
        </>
    );
}
