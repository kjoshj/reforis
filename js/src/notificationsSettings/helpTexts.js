/*
 * Copyright (C) 2019-2025 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

const HELP_TEXTS = {
    smtp_type: _(
        'If you set SMTP provider to "Turris", the servers provided to members of the Turris project would be ' +
            "used. These servers do not require any additional settings. If you want to set your own SMTP server, please " +
            'select "Custom" and enter required settings.'
    ),
    common: {
        severity_filter: _(
            "Depending on the importance or severity of the notification, you can choose to receive notifications about all events or only about those that require your attention."
        ),
        to: _(
            "Email address of recipient. Separate multiple addresses by comma."
        ),
        send_news: _("Send notifications about new releases."),
    },
    smtp_turris: {
        sender_name: _(
            'Name of the sender that will be used before the "at" symbol in the email address. For example: "home-router@notify.turis.cz"'
        ),
    },
    smtp_custom: {
        from: _("This is the address notifications are send from."),
    },
};

export default HELP_TEXTS;
