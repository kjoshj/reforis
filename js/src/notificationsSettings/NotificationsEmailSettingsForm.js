/*
 * Copyright (C) 2019-2025 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import { Switch, RadioSet } from "foris";
import PropTypes from "prop-types";

import CommonForm from "./CommonForm";
import HELP_TEXTS from "./helpTexts";
import SMTPCustomForm from "./SMTPCustomForm";
import SMTPTurrisForm from "./SMTPTurrisForm";

const SMTP_TYPE_CHOICES = [
    {
        label: _("Turris"),
        value: "turris",
    },
    {
        label: _("Custom"),
        value: "custom",
    },
];

NotificationsEmailSettingsForm.propTypes = {
    formData: PropTypes.shape({
        emails: PropTypes.shape({
            enabled: PropTypes.bool,
            smtp_type: PropTypes.oneOf(["turris", "custom"]),
            common: PropTypes.object,
            smtp_turris: PropTypes.object,
            smtp_custom: PropTypes.object,
        }),
    }).isRequired,
    formErrors: PropTypes.shape({
        smtp_turris: PropTypes.object,
        smtp_custom: PropTypes.object,
        common: PropTypes.object,
    }),
    setFormValue: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
};

NotificationsEmailSettingsForm.defaultProps = {
    setFormValue: () => {},
    formData: {},
    formErrors: {},
};

export default function NotificationsEmailSettingsForm({
    formData,
    formErrors,
    setFormValue,
    disabled,
}) {
    const { emails: emailsFormData } = formData;

    let smtpForm = null;
    if (emailsFormData.smtp_type === "turris") {
        smtpForm = (
            <SMTPTurrisForm
                formData={emailsFormData.smtp_turris}
                formErrors={formErrors.smtp_turris}
                setFormValue={setFormValue}
                disabled={disabled}
            />
        );
    } else if (emailsFormData.smtp_type === "custom") {
        smtpForm = (
            <SMTPCustomForm
                formData={emailsFormData.smtp_custom}
                formErrors={formErrors.smtp_custom}
                setFormValue={setFormValue}
                disabled={disabled}
            />
        );
    }

    return (
        <>
            <h2>{_("Email Notification Settings")}</h2>
            <Switch
                label={_("Enable email notifications")}
                checked={emailsFormData.enabled}
                onChange={setFormValue((value) => ({
                    emails: {
                        enabled: { $set: value },
                    },
                }))}
                disabled={disabled}
            />
            {emailsFormData.enabled && (
                <>
                    <RadioSet
                        label={_("SMTP provider")}
                        name="smtp_provider"
                        choices={SMTP_TYPE_CHOICES}
                        value={emailsFormData.smtp_type}
                        helpText={HELP_TEXTS.smtp_type}
                        inline
                        onChange={setFormValue((value) => ({
                            emails: {
                                smtp_type: { $set: value },
                            },
                        }))}
                        disabled={disabled}
                    />
                    <CommonForm
                        formData={emailsFormData.common}
                        formErrors={formErrors.common}
                        setFormValue={setFormValue}
                        disabled={disabled}
                    />
                    {smtpForm}
                </>
            )}
        </>
    );
}
