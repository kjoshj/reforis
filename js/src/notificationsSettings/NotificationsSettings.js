/*
 * Copyright (C) 2019-2025 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { ForisForm } from "foris";
import PropTypes from "prop-types";

import API_URLs from "common/API";

import NotificationsCommonSettingsForm from "./NotificationsSettingsCommonForm";
import NotificationsEmailSettingsForm from "./NotificationsSettingsEmailForm";
import NotificationsPushSettingsForm from "./NotificationsSettingsPushForm";
import TestNotification from "./TestNotification";
import validator from "./validator";

NotificationsSettings.propTypes = {
    ws: PropTypes.object.isRequired,
};

export default function NotificationsSettings({ ws }) {
    return (
        <>
            <h1>{_("Notifications")}</h1>
            <p>
                {_(
                    "Your device can send you e-mail notifications about required reboots, important events, installed updates, and new features according to your settings. Once you are done with the configuration, you can test it by pressing Send test notification. Additionally, you can use "
                )}
                <a
                    href="https://ntfy.sh/"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    {_("ntfy")}
                    <sup>
                        <FontAwesomeIcon
                            icon={["fas", "external-link-alt"]}
                            className="fa-sm ms-1"
                        />
                    </sup>
                </a>
                {_(
                    " to receive real-time push notifications directly to your mobile device or desktop. ntfy allows you to subscribe to topics and get instant alerts without the need for email."
                )}
            </p>
            <ForisForm
                ws={ws}
                forisConfig={{
                    endpoint: API_URLs.notificationsSettings,
                    wsModule: "router_notifications",
                    wsAction: "update_email_settings",
                }}
                prepData={prepData}
                prepDataToSubmit={prepDataToSubmit}
                validator={validator}
            >
                <NotificationsEmailSettingsForm />
                <NotificationsPushSettingsForm />
                <NotificationsCommonSettingsForm />
                <TestNotification />
            </ForisForm>
            <div id="test-notification" />
        </>
    );
}

function prepData(formData) {
    formData.emails.common.to = formData.emails.common.to.join(", ");
    return formData;
}

function prepDataToSubmit(formData) {
    // Emails
    if (formData.emails.enabled) {
        formData.emails.common.to = formData.emails.common.to
            .replace(/\s+/g, "")
            .split(",");

        if (formData.emails.smtp_type === "turris")
            delete formData.emails.smtp_custom;
        else if (formData.emails.smtp_type === "custom")
            delete formData.emails.smtp_turris;
    } else {
        formData.emails = {
            enabled: false,
        };
    }

    // Ntfy
    if (formData.ntfy.enabled) {
        formData.ntfy.url = formData.ntfy.url.trim();
    } else {
        formData.ntfy = {
            enabled: false,
        };
    }

    // Reboots
    delete formData.reboots;

    return formData;
}
