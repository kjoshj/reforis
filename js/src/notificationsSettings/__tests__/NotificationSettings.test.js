/*
 * Copyright (C) 2020-2025 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";

import {
    fireEvent,
    getByLabelText,
    getByText,
    render,
    wait,
    waitForElement,
} from "foris/testUtils/customTestRender";

import { WebSockets } from "foris";
import { mockJSONError } from "foris/testUtils/network";
import mockAxios from "jest-mock-axios";
import notificationsSettings, {
    testNotificationSeverityFixture2,
    testNotificationSeverityFixture3,
} from "./__fixtures__/notificationsSettings";

import NotificationsSettings from "../NotificationsSettings";
import {
    UNSAVED_CHANGES_MODAL_MESSAGE,
    SEVERITY_ALERT_MESSAGE,
} from "../TestNotification";

const ENABLE_CHECKBOX_LABEL = "Enable email notifications";

describe("<NotificationsSettings/>", () => {
    let NotificationsSettingsContainer;

    beforeEach(async () => {
        const webSockets = new WebSockets();
        const { container } = render(
            <>
                <NotificationsSettings ws={webSockets} />
                <div id="modal-container" />
            </>
        );
        mockAxios.mockResponse({ data: notificationsSettings() });
        NotificationsSettingsContainer = container;
        await wait(() => {
            expect(
                getByLabelText(container, ENABLE_CHECKBOX_LABEL)
            ).toBeTruthy();
        });
    });

    it("should handle error", async () => {
        const webSockets = new WebSockets();
        const { container } = render(<NotificationsSettings ws={webSockets} />);
        mockJSONError();
        await wait(() =>
            getByText(container, "An error occurred while fetching data.")
        );
    });

    describe("Email notifications", () => {
        it("Enabled, smtp_type:custom", () => {
            expect(NotificationsSettingsContainer).toMatchSnapshot();
        });

        it("Disabled", () => {
            fireEvent.click(
                getByLabelText(
                    NotificationsSettingsContainer,
                    ENABLE_CHECKBOX_LABEL
                )
            );
            expect(NotificationsSettingsContainer).toMatchSnapshot();
        });

        it("Enabled,smtp_type:turris", () => {
            fireEvent.click(
                getByLabelText(NotificationsSettingsContainer, "Turris")
            );
            expect(NotificationsSettingsContainer).toMatchSnapshot();
        });

        it("Post.", () => {
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toBeCalled();
            const data = {
                emails: {
                    common: {
                        send_news: true,
                        severity_filter: 1,
                        to: ["some@example.com"],
                    },
                    enabled: true,
                    smtp_custom: {
                        from: "router@example.com",
                        host: "example.com",
                        password: "test_password",
                        port: 465,
                        security: "ssl",
                        username: "root",
                    },
                    smtp_type: "custom",
                },
                ntfy: {
                    enabled: false,
                },
            };
            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                data,
                expect.anything()
            );
        });

        it("Post smtp_type:turris.", () => {
            fireEvent.click(
                getByLabelText(NotificationsSettingsContainer, "Turris")
            );
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toBeCalled();
            const data = {
                emails: {
                    common: {
                        send_news: true,
                        severity_filter: 1,
                        to: ["some@example.com"],
                    },
                    enabled: true,
                    smtp_type: "turris",
                    smtp_turris: { sender_name: "turris" },
                },
                ntfy: {
                    enabled: false,
                },
            };
            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                data,
                expect.anything()
            );
        });

        it("Post disabled.", () => {
            fireEvent.click(
                getByLabelText(
                    NotificationsSettingsContainer,
                    ENABLE_CHECKBOX_LABEL
                )
            );
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toBeCalled();
            const data = {
                emails: { enabled: false },
                ntfy: {
                    enabled: false,
                },
            };
            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                data,
                expect.anything()
            );
        });
    });

    describe("ntfy", () => {
        it("Disabled", () => {
            expect(NotificationsSettingsContainer).toMatchSnapshot();
        });

        it("Enabled", () => {
            fireEvent.click(
                getByLabelText(
                    NotificationsSettingsContainer,
                    "Enable ntfy notifications"
                )
            );
            expect(NotificationsSettingsContainer).toMatchSnapshot();
        });

        it("Post.", () => {
            fireEvent.click(
                getByLabelText(
                    NotificationsSettingsContainer,
                    "Enable ntfy notifications"
                )
            );

            fireEvent.input(
                getByLabelText(NotificationsSettingsContainer, "URL"),
                { target: { value: "https://example.com" } }
            );

            fireEvent.change(
                getByLabelText(NotificationsSettingsContainer, "Priority"),
                { target: { value: "max" } }
            );

            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            const data = {
                emails: {
                    common: {
                        send_news: true,
                        severity_filter: 1,
                        to: ["some@example.com"],
                    },
                    enabled: true,
                    smtp_custom: {
                        from: "router@example.com",
                        host: "example.com",
                        password: "test_password",
                        port: 465,
                        security: "ssl",
                        username: "root",
                    },
                    smtp_type: "custom",
                },
                ntfy: {
                    enabled: true,
                    url: "https://example.com",
                    priority: "max",
                },
            };
            expect(mockAxios.post).toBeCalled();
            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                data,
                expect.anything()
            );
        });
    });

    describe("<TestNotification/>", () => {
        let option;

        beforeEach(() => {
            option = getByLabelText(
                NotificationsSettingsContainer,
                /Importance/
            );
        });

        it("Displaying test notification importance warning (severity: 1)", () => {
            expect(
                NotificationsSettingsContainer.childNodes[3].childNodes[0]
                    .innerHTML
            ).toEqual(SEVERITY_ALERT_MESSAGE);
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });

        it("No test notification importance warning (severity: 2)", () => {
            fireEvent.change(option, {
                target: {
                    value: "2",
                },
            });
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                testNotificationSeverityFixture2(),
                expect.anything()
            );
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });

        it("No test notification importance warning (severity: 3)", () => {
            fireEvent.change(option, {
                target: {
                    value: "3",
                },
            });
            fireEvent.click(getByText(NotificationsSettingsContainer, "Save"));

            expect(mockAxios.post).toHaveBeenCalledWith(
                "/reforis/api/notifications-settings",
                testNotificationSeverityFixture3(),
                expect.anything()
            );
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });

        it("Should toggle unsaved changes warning modal", async () => {
            fireEvent.change(option, {
                target: {
                    value: "2",
                },
            });
            fireEvent.click(
                getByText(
                    NotificationsSettingsContainer,
                    "Send test notification"
                )
            );
            await waitForElement(() =>
                getByText(
                    NotificationsSettingsContainer,
                    UNSAVED_CHANGES_MODAL_MESSAGE
                )
            );
            expect(
                NotificationsSettingsContainer.childNodes[3]
            ).toMatchSnapshot();
        });
    });
});
