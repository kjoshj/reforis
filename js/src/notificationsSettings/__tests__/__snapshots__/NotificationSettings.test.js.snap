// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationsSettings/> <TestNotification/> Displaying test notification importance warning (severity: 1) 1`] = `
<div
  id="test-notification"
>
  <div
    class="alert alert-warning alert-fade-in"
    role="alert"
  >
    You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
  </div>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      Test Notification
    </h2>
    <p>
      Here you can verify whether SMTP is configured correctly by sending a test notification to your email inbox.
    </p>
    <div
      class="text-end"
    >
      <button
        class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
        type="button"
      >
        <span>
          Send test notification
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<NotificationsSettings/> <TestNotification/> No test notification importance warning (severity: 2) 1`] = `
<div
  id="test-notification"
>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      Test Notification
    </h2>
    <p>
      Here you can verify whether SMTP is configured correctly by sending a test notification to your email inbox.
    </p>
    <div
      class="text-end"
    >
      <button
        class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
        type="button"
      >
        <span>
          Send test notification
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<NotificationsSettings/> <TestNotification/> No test notification importance warning (severity: 3) 1`] = `
<div
  id="test-notification"
>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      Test Notification
    </h2>
    <p>
      Here you can verify whether SMTP is configured correctly by sending a test notification to your email inbox.
    </p>
    <div
      class="text-end"
    >
      <button
        class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
        type="button"
      >
        <span>
          Send test notification
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<NotificationsSettings/> <TestNotification/> Should toggle unsaved changes warning modal 1`] = `
<div
  id="test-notification"
>
  <div
    class="alert alert-warning alert-fade-in"
    role="alert"
  >
    You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
  </div>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <h2>
      Test Notification
    </h2>
    <p>
      Here you can verify whether SMTP is configured correctly by sending a test notification to your email inbox.
    </p>
    <div
      class="text-end"
    >
      <button
        class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
        type="button"
      >
        <span>
          Send test notification
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<NotificationsSettings/> Disabled 1`] = `
<div>
  <h1>
    Notifications
  </h1>
  <p>
    Your device can send you e-mail notifications about required reboots, important events, installed updates and new features according to your settings. When done with configuration, you can test it by pressing Send test notification.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        Notification Settings
      </h2>
      <div
        class="form-check form-switch mb-3"
      >
        <input
          checked=""
          class="form-check-input"
          id="1"
          role="switch"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="1"
        >
          Enable Email Notifications
        </label>
      </div>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          type="submit"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </form>
  </div>
  <div
    id="test-notification"
  />
  <div
    id="modal-container"
  />
</div>
`;

exports[`<NotificationsSettings/> Enabled, smtp_type:custom 1`] = `
<div>
  <h1>
    Notifications
  </h1>
  <p>
    Your device can send you e-mail notifications about required reboots, important events, installed updates and new features according to your settings. When done with configuration, you can test it by pressing Send test notification.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        Notification Settings
      </h2>
      <div
        class="form-check form-switch mb-3"
      >
        <input
          checked=""
          class="form-check-input"
          id="1"
          role="switch"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="1"
        >
          Enable Email Notifications
        </label>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="d-block"
          for="2"
        >
          SMTP provider
        </label>
        <div
          class="mb-3 form-check form-check-inline"
        >
          <input
            class="form-check-input me-2"
            id="smtp_provider-0"
            name="smtp_provider"
            type="radio"
            value="turris"
          />
          <label
            class="form-check-label"
            for="smtp_provider-0"
          >
            Turris
          </label>
        </div>
        <div
          class="mb-3 form-check form-check-inline"
        >
          <input
            checked=""
            class="form-check-input me-2"
            id="smtp_provider-1"
            name="smtp_provider"
            type="radio"
            value="custom"
          />
          <label
            class="form-check-label"
            for="smtp_provider-1"
          >
            Custom
          </label>
        </div>
        <div
          class="form-text"
        >
          <small>
            If you set SMTP provider to "Turris", the servers provided to members of the Turris project would be used. These servers do not require any additional settings. If you want to set your own SMTP server, please select "Custom" and enter required settings.
          </small>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="3"
        >
          Recipient's email
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="3"
            required=""
            type="text"
            value="some@example.com"
          />
        </div>
        <div
          class="form-text"
        >
          <small>
            Email address of recipient. Separate multiple addresses by comma.
          </small>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="4"
        >
          Importance
        </label>
        <select
          class="form-select"
          id="4"
        >
          <option
            value="1"
          >
            Reboot is required
          </option>
          <option
            value="2"
          >
            Reboot or attention is required
          </option>
          <option
            value="3"
          >
            Reboot or attention is required or update was installed
          </option>
        </select>
      </div>
      <div
        class="mb-3 form-check"
      >
        <input
          checked=""
          class="form-check-input"
          id="5"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="5"
        >
          Send news
        </label>
        <div
          class="form-text"
        >
          <small>
            Send emails about new features.
          </small>
        </div>
      </div>
      <h3>
        SMTP Settings
      </h3>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="6"
        >
          Sender address (From)
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="6"
            placeholder="router@example.com"
            required=""
            type="email"
            value="router@example.com"
          />
        </div>
        <div
          class="form-text"
        >
          <small>
            This is the address notifications are send from.
          </small>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="7"
        >
          Server address
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="7"
            placeholder="example.com"
            required=""
            type="text"
            value="example.com"
          />
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="8"
        >
          Server port
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="8"
            max="65535"
            min="1"
            type="number"
            value="465"
          />
          <button
            aria-label="Increase value. Hint: Hold to increase faster."
            class="btn btn-outline-secondary"
            title="Increase value. Hint: Hold to increase faster."
            type="button"
          >
            <i
              class="fa"
            />
          </button>
          <button
            aria-label="Decrease value. Hint: Hold to decrease faster."
            class="btn btn-outline-secondary"
            title="Decrease value. Hint: Hold to decrease faster."
            type="button"
          >
            <i
              class="fa"
            />
          </button>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="9"
        >
          Security
        </label>
        <select
          class="form-select"
          id="9"
        >
          <option
            value="none"
          >
            None
          </option>
          <option
            value="ssl"
          >
            SSL
          </option>
          <option
            value="starttls"
          >
            STARTTLS
          </option>
        </select>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="10"
        >
          Username
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="10"
            type="text"
            value="root"
          />
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="11"
        >
          Password
        </label>
        <div
          class="input-group"
        >
          <input
            autocomplete="current-password"
            class="form-control"
            id="11"
            type="password"
            value="test_password"
          />
          <button
            class="input-group-text"
            type="button"
          >
            <i
              class="fa"
            />
          </button>
        </div>
      </div>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          type="submit"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </form>
  </div>
  <div
    id="test-notification"
  >
    <div
      class="alert alert-warning alert-fade-in"
      role="alert"
    >
      You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
    </div>
    <div
      class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
    >
      <h2>
        Test Notification
      </h2>
      <p>
        Here you can verify whether SMTP is configured correctly by sending a test notification to your email inbox.
      </p>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          type="button"
        >
          <span>
            Send test notification
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    id="modal-container"
  >
    <div
      aria-labelledby="modal-title"
      aria-modal="true"
      class="modal fade"
      role="dialog"
    >
      <div
        class="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h1
              class="modal-title fs-5"
            >
              Warning!
            </h1>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            <p>
              There are some unsaved changes in the notifications settings. Do you want to discard them and test the notifications with the old settings?
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary d-inline-flex justify-content-center align-items-center"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="btn btn-danger d-inline-flex justify-content-center align-items-center"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NotificationsSettings/> Enabled,smtp_type:turris 1`] = `
<div>
  <h1>
    Notifications
  </h1>
  <p>
    Your device can send you e-mail notifications about required reboots, important events, installed updates and new features according to your settings. When done with configuration, you can test it by pressing Send test notification.
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        Notification Settings
      </h2>
      <div
        class="form-check form-switch mb-3"
      >
        <input
          checked=""
          class="form-check-input"
          id="1"
          role="switch"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="1"
        >
          Enable Email Notifications
        </label>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="d-block"
          for="2"
        >
          SMTP provider
        </label>
        <div
          class="mb-3 form-check form-check-inline"
        >
          <input
            class="form-check-input me-2"
            id="smtp_provider-0"
            name="smtp_provider"
            type="radio"
            value="turris"
          />
          <label
            class="form-check-label"
            for="smtp_provider-0"
          >
            Turris
          </label>
        </div>
        <div
          class="mb-3 form-check form-check-inline"
        >
          <input
            checked=""
            class="form-check-input me-2"
            id="smtp_provider-1"
            name="smtp_provider"
            type="radio"
            value="custom"
          />
          <label
            class="form-check-label"
            for="smtp_provider-1"
          >
            Custom
          </label>
        </div>
        <div
          class="form-text"
        >
          <small>
            If you set SMTP provider to "Turris", the servers provided to members of the Turris project would be used. These servers do not require any additional settings. If you want to set your own SMTP server, please select "Custom" and enter required settings.
          </small>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="3"
        >
          Recipient's email
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="3"
            required=""
            type="text"
            value="some@example.com"
          />
        </div>
        <div
          class="form-text"
        >
          <small>
            Email address of recipient. Separate multiple addresses by comma.
          </small>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="4"
        >
          Importance
        </label>
        <select
          class="form-select"
          id="4"
        >
          <option
            value="1"
          >
            Reboot is required
          </option>
          <option
            value="2"
          >
            Reboot or attention is required
          </option>
          <option
            value="3"
          >
            Reboot or attention is required or update was installed
          </option>
        </select>
      </div>
      <div
        class="mb-3 form-check"
      >
        <input
          checked=""
          class="form-check-input"
          id="5"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="5"
        >
          Send news
        </label>
        <div
          class="form-text"
        >
          <small>
            Send emails about new features.
          </small>
        </div>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="12"
        >
          Sender's name
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="12"
            type="text"
            value="turris"
          />
        </div>
        <div
          class="form-text"
        >
          <small>
            Name of the sender - will be used as a part of the sender's email address before the "at" sign.
          </small>
        </div>
      </div>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          type="submit"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </form>
  </div>
  <div
    id="test-notification"
  >
    <div
      class="alert alert-warning alert-fade-in"
      role="alert"
    >
      You will not receive the test notification to your email inbox with current importance level. Please raise the importance level at least to "Reboot or attention is required" and click "Save" button if you want to get this notification by email.
    </div>
    <div
      class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
    >
      <h2>
        Test Notification
      </h2>
      <p>
        Here you can verify whether SMTP is configured correctly by sending a test notification to your email inbox.
      </p>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          type="button"
        >
          <span>
            Send test notification
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    id="modal-container"
  >
    <div
      aria-labelledby="modal-title"
      aria-modal="true"
      class="modal fade"
      role="dialog"
    >
      <div
        class="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h1
              class="modal-title fs-5"
            >
              Warning!
            </h1>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            <p>
              There are some unsaved changes in the notifications settings. Do you want to discard them and test the notifications with the old settings?
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary d-inline-flex justify-content-center align-items-center"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="btn btn-danger d-inline-flex justify-content-center align-items-center"
              type="button"
            >
              <span>
                Confirm
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
